accounting_reporting_core/src/main/java/org/cardanofoundation/lob/app/accounting_reporting_core/repository/CustomTransactionRepository.java

```diff
@@ -2,12 +2,24 @@
 
 import org.cardanofoundation.lob.app.accounting_reporting_core.domain.core.TransactionType;
 import org.cardanofoundation.lob.app.accounting_reporting_core.domain.core.ValidationStatus;
+import org.cardanofoundation.lob.app.accounting_reporting_core.domain.entity.ReconcilationRejectionCode;
 import org.cardanofoundation.lob.app.accounting_reporting_core.domain.entity.TransactionEntity;
+import org.cardanofoundation.lob.app.accounting_reporting_core.resource.requests.ReconciliationFilterRequest;
+import org.cardanofoundation.lob.app.accounting_reporting_core.resource.requests.ReconciliationFilterStatusRequest;
 
 import java.util.List;
+import java.util.Set;
 
 public interface CustomTransactionRepository {
     List<TransactionEntity> findAllByStatus(String organisationId,
                                             List<ValidationStatus> validationStatuses,
                                             List<TransactionType> transactionType);
+
+    public List<Object[]> findAllReconciliationSpecial(ReconciliationFilterRequest body);
+
+    public List<Object[]> findAllReconciliationSpecialCount(ReconciliationFilterRequest body);
+
+    public List<TransactionEntity> findAllReconciliation(ReconciliationFilterRequest body);
+
+    public Object[] findCalcReconciliationStatistic();
```

Comment:
Why are you returning Object, this is anti-pattern in java, we never do this.

Reply:
Is because the query return different fields with a join from different tables and we don't have any dynamic/object that fill that query.
That's why is an object and not a mapped/specific object.

Reply:
We discussed it on our pair programming session and found the solution(s) for perhaps different refactoring tickets.